src/riak_test_escript.erl

```diff
@@ -326,20 +326,25 @@ run_test(Test, TestType, Outdir, TestMetaData, Report, HarnessArgs, NumTests) ->
                                                   zlib:gzip(element(2,file:read_file(CoverageFile)))}) || CoverageFile /= cover_disabled ],
                     ResultPlusGiddyUp = TestResult ++ [{giddyup_url, list_to_binary(Base)}],
                     [ rt:post_result(ResultPlusGiddyUp, WebHook) || WebHook <- get_webhooks() ],
-                    CTLogTarFile = "/tmp/ctlogs_" ++ integer_to_list(erlang:phash2(make_ref())),
-                    case erl_tar:create(CTLogTarFile, ["ct_logs"], [compressed]) of
-                        ok ->
-                            giddyup:post_artifact(Base, {"ct_logs.tar.gz",
-                                element(2,file:read_file(CTLogTarFile))});
-                            file:delete(CTLogTarFile);
-                        %% We can't load them for whatever reason so skip
-                        _ -> ok
-                    end
+                    archive_ct_logs_to_giddyup(Base)
             end
     end,
     rt_cover:stop(),
     [{coverdata, CoverageFile} | SingleTestResult].
 
+archive_ct_logs_to_giddyup(Base) ->
+    CTLogTarFile = "/tmp/ctlogs_" ++ integer_to_list(erlang:phash2(make_ref())),
+    Result = erl_tar:create(CTLogTarFile, ["ct_logs"], [compressed]),
+    maybe_post_ct_to_giddyup(Result, Base, CTLogTarFile).
+
+maybe_post_ct_to_giddyup(ok, Base, CTLogTarFile) ->
+    giddyup:post_artifact(Base, {"ct_logs.tar.gz",
+                          element(2,file:read_file(CTLogTarFile))}),
```

Comment:
If `CTLogTarFile` cannot be read, the second element in the tuple returned by `file:read_file/1` would be an atom. I know it's unlikely, but in such a case `giddyup:post_artifact/2` will be called with that atom (and not the file contents) entupled with the file name, most likely causing a badarg exception, and somewhere far far away, one Robert Virding will cry.